### PR TITLE
EN-80: Set and see vacations per employee per year

### PR DESCRIPTION
**Pull request by FacuOrellana**
# Description :pencil:

New vacations entity and table. Permissions updated for dev, HR manager and admin role.

## Link to ticket :link:

https://entropyteam.atlassian.net/browse/EN-80

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? :microscope:
Manual testing

Vacation table
image

successful migrations
image

### CODE REVIEW

File: src/main/java/com/entropyteam/entropay/employees/controllers/VacationController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.UUID;
+
+import static com.entropyteam.entropay.auth.AuthConstants.ROLE_MANAGER_HR;
+
+@RestController
+@CrossOrigin
+@Secured({ROLE_MANAGER_HR})
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/controllers/VacationController.java:18]: Let's allow Admins and developers to use this endpoint

File: src/main/java/com/entropyteam/entropay/employees/dtos/VacationDto.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+public record VacationDto(  UUID id,
+                            @NotNull
+                            String year,
+                            Integer credit,
+                            Integer debit,
```
Comment (agustinb4) [inline src/main/java/com/entropyteam/entropay/employees/dtos/VacationDto.java:14]: Should we check for null?

**Review by agustinb4 — CHANGES_REQUESTED** [review]

**Review by agustinb4 — APPROVED** [review]
Looks good!

**Comment by FedericoA15** [comment]
good job

**Pull request merged**